pylint/checkers/base/name_checker/checker.py

```diff
@@ -298,7 +300,9 @@ def _create_naming_rules(self) -> Tuple[Dict[str, Pattern[str]], Dict[str, str]]
                 regexps[name_type] = DEFAULT_PATTERNS[name_type]
 
             custom_regex_setting_name = f"{name_type}_rgx"
-            custom_regex = getattr(self.config, custom_regex_setting_name, None)
+            custom_regex = getattr(
+                self.linter.namespace, custom_regex_setting_name, None
+            )
```

Comment:
Should we handle this in the option parsing directly and not later with getattr ? I think it's not ideal as it's a delayed default value that put a little of option parsing inside checkers.

Reply:
These options are created dynamically for each name type we have.

We can probably define them in `checker.options` but that's more a general refactor of this checker. By creating them dynamically we do remove a lot of duplication of the code though.